rapidjson.cpp

```diff
@@ -4067,11 +4103,8 @@ module_exec(PyObject* m)
         )
         return -1;
 
-    Py_INCREF(&Decoder_Type);
-    if (PyModule_AddObject(m, "Decoder", (PyObject*) &Decoder_Type) < 0) {
-        Py_DECREF(&Decoder_Type);
+    if (PyModule_AddObject(m, "Decoder", decoder_type.get()) < 0)
```

Comment:
Using RAII here prolongates lifetime of a strong reference to decoder type to the end of the module exec function. It shouldn't affect type object lifetime since there are other strong references to it. But using RAII make the code smaller and less error prone.